hudi-common/src/main/java/org/apache/hudi/common/util/FileIOUtils.java

```diff
@@ -204,4 +204,40 @@ public static Option<byte[]> readDataFromPath(FileSystem fileSystem, org.apache.
   public static Option<byte[]> readDataFromPath(FileSystem fileSystem, org.apache.hadoop.fs.Path detailPath) {
     return readDataFromPath(fileSystem, detailPath, false);
   }
+
+  /**
+   * Return the configured local directories where hudi can write files. This
+   * method does not create any directories on its own, it only encapsulates the
+   * logic of locating the local directories according to deployment mode.
+   */
+  public static String[] getConfiguredLocalDirs() {
+    if (isRunningInYarnContainer()) {
+      // If we are in yarn mode, systems can have different disk layouts so we must set it
```

Comment:
We totally ignore the option through hoodie write config then ?

Reply:
> We totally ignore the option through hoodie write config then ?

First, get the local dir from the yarn container. If you can't get it, you can get it from the hoodie write config. In this way, users on yarn basically do not need to actively modify the hoodie write config.

Reply:
> If you can't get it, you can get it from the hoodie write config

I didn't see the logic.

Reply:
> > If you can't get it, you can get it from the hoodie write config
> 
> I didn't see the logic.

https://github.com/apache/hudi/blob/0946f43b44aae6a01ecc6a610d9bd48404aaaacd/hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieMergedLogRecordScanner.java#L98

Reply:
Thanks, should we take the write config as higher priority then ? If user configured it explicitly.

Reply:
> Thanks, should we take the write config as higher priority then ? If user configured it explicitly.

It may be necessary to remove the default value
```
  public static final ConfigProperty<String> SPILLABLE_MAP_BASE_PATH = ConfigProperty
      .key("hoodie.memory.spillable.map.path")
      .defaultValue("/tmp/")
      .withDocumentation("Default file path prefix for spillable map");
```

Reply:
> Thanks, should we take the write config as higher priority then ? If user configured it explicitly.

   It may be necessary to remove the default value
   ```
     public static final ConfigProperty<String> SPILLABLE_MAP_BASE_PATH = ConfigProperty
         .key("hoodie.memory.spillable.map.path")
         .defaultValue("/tmp/")
         .withDocumentation("Default file path prefix for spillable map");
   ```